test_task_exists.py

```diff
@@ -0,0 +1,54 @@
+from todo import TodoList
+from pytest import fixture
+
+@fixture
+def empty_todolist():
+    return TodoList()
+
+@fixture
+def populated_todolist():
+    todo_list = TodoList()
+    todo_list.tasks = [
+        ("Buy grocery", "Medium", "2024-12-02", [], "Long", False),
+        ("Call doctor", "High", "2024-11-12", [], "Long", False),
+        ("Finish homework", "High", None, ["school"], "Long", False)
+    ]
+    return todo_list
+
+def test_task_exists_found(populated_todolist):
+    """Check if an existing task is found by task_exists."""
+    assert populated_todolist.task_exists("Buy grocery") == 1
+
+def test_task_exists_not_found(populated_todolist):
+    """Ensure task_exists returns None if the task is not in the list."""
+    assert populated_todolist.task_exists("Go to gym") is None
+
+def test_task_exists_case_insensitivity(populated_todolist):
+    """Verify that task_exists is case-insensitive."""
+    assert populated_todolist.task_exists("buy GROCERY") == 1
+
+def test_task_exists_whitespace_handling(populated_todolist):
+    """Check if task_exists handles leading and trailing whitespace correctly."""
+    assert populated_todolist.task_exists("  Call doctor  ") == 2
+
+def test_task_exists_empty_list(empty_todolist):
+    """Test task_exists with an empty task list, expecting None."""
+    assert empty_todolist.task_exists("Read book") is None
+
+def test_task_exists_empty_string(populated_todolist):
+    """Verify that an empty string as a task name returns None."""
+    assert populated_todolist.task_exists("") is None
+
+def test_task_exists_task_with_only_whitespace(populated_todolist):
+    """Ensure task_exists returns None when searching for a name consisting only of whitespace."""
+    assert populated_todolist.task_exists("   ") is None
```

Comment:
Should expect return of -1, not None according to documentation

Reply:
def task_exists(self, task):
        """Checks if a task already exists. Returns the task number if it exists, or None otherwise.

        Args:
            task (_type_): a string

        Returns:
            _type_: None if task does not exist. Int if task exists.
        """
        task = task.strip()

        for idx, t in enumerate(self.tasks):
            task_name = t[0].strip().lower()
            if task_name == task.lower():
                return idx + 1

        return None  # Task does not exist
The function doesnt return -1